src/components/network-chart/index.js

```diff
@@ -50,27 +47,76 @@ type Props = {|
   ...ConnectedProps<{||}, StateProps, DispatchProps>,
 |};
 
-/*
- * The VirtualListRows only re-render when their items have changed. This information
- * is derived from the current props, which includes time range, container sizing,
- * and the marker information itself. In order to properly update the rows,
- * provide a memoized function that can compute this on the fly.
- */
-const _getVirtualListItemsMemoized = memoize(_getVirtualListItems, {
-  cache: new MixedTupleMap(),
-});
-
 class NetworkChart extends React.PureComponent<Props> {
+  // This isn't used at the moment, but we need a fixed instance so that a
+  // rerender isn't triggered in VirtualList.
+  _specialItems = [];
+
   _onCopy = (_event: Event) => {
-    // No implemented.
+    // Not implemented.
   };
 
   _onKeyDown = (_event: KeyboardEvent) => {
-    // No implemented.
+    // Not implemented.
+  };
+
+  /**
+   * Convert the time for a network marker into the CssPixels to be used on the screen.
+   * This function takes into account the range used, as well as the container sizing
+   * as passed in by the WithSize component.
+   */
+  _timeToCssPixels(time: Milliseconds): CssPixels {
+    const { timeRange, width } = this.props;
+    const timeRangeTotal = timeRange.end - timeRange.start;
+    const innerContainerWidth =
+      width - TIMELINE_MARGIN_LEFT - TIMELINE_MARGIN_RIGHT;
+
+    const markerPosition =
+      ((time - timeRange.start) / timeRangeTotal) * innerContainerWidth +
+      TIMELINE_MARGIN_LEFT;
+
+    return markerPosition;
+  }
+
+  _renderRow = (marker: Marker, index: number): React.Node => {
+    const { threadIndex } = this.props;
```

Comment:
I didn't change the logic here; I merely moved the code from the previous `_getVirtualListItems`. The good thing is that now the computation is done only for these items that are rendered.